### ISSUE TITLE
runtime policy with hashes having the \ prefix 

### ISSUE THREAD

**Issue by kkaarreell** [keylime/keylime#1466]
Is your issue a feature request? If so, please raise it as an [enhancement](https://github.com/keylime/enhancements)

# Environment
 * OS / version: CentOS Stream 9
 * Processor architecture: x86_64
 * TPM Manufacturer: swtpm
 * Keylime version: 7.3.0

# Description 
The current script `create_runtime_policy.sh` uses `shaXXXsum` functions to compute file digest. However, this script may add `\` before the digest in some cases. 
```
$ echo foo > ba\\r
$ sha256sum ba\\r 
\b5bb9d8014a0f9b1d61e21e796d78dccdf1352f23cd32812f4850b878ae4944c  ba\\r
```
This is intentional, see e.g. https://bugzilla.redhat.com/show_bug.cgi?id=1882187

When trying to use such a policy I am getting similar errors
```
2023-09-14 03:10:43.085 - keylime.tenant - ERROR - Response code 400: Runtime policy is malformatted: '\\ac9e9eb8981bcf841e1b6341e4d470c1d1dba396a16f6221840cb358b974d815' does not match '^[0-9a-f]{40,128}$'
```
Either keylime should accept and strip `\\` before the digest (I would suggest) or keylime tools should be updated not to add this prefix (Either using the `--zero` option for `sha256sum` or by other means).

**Comment by stefanberger** [comment]
create_runtime_policy.sh already contains some cleanup code that may be unnecessary or wrong:

```
# A bit of cleanup on the resulting file (among other problems, sha256sum might output a hash with the prefix '\\')
sed -i "s/^\\\//g" $ALLOWLIST_DIR/${OUTPUT}
```

Does adding this line  here help?

```
sed -i 's/^\\//' $ALLOWLIST_DIR/${OUTPUT}
```

**Comment by kkaarreell** [comment]
Yes, the later sed is what I did, the first one is incorrect I guess.

**Comment by stefanberger** [comment]
> Yes, the later sed is what I did, the first one is incorrect I guess.

You did the change already?  Are you going to send a PR for it? I think it's the right place to fix this issue in the tool that creates the policy...

**Comment by kkaarreell** [comment]
I just did this allowlist file update locally. Yes, I can prepare PR.

**Comment by kkaarreell** [comment]
OK, so now I have noticed that the script has changed in upstream in the meantime. I am clearly using its older version.
I have tested the current version and I can confirm it works, i.e.
```
sed -i "s/^\\\//g" $ALLOWLIST_DIR/${OUTPUT}
```
does its job.
The other expression `'s/^\\//'` is identical as avoiding double-quotes lets us save one `\`.

So the reported issue does not apply to the current upstream version. It's just unfortunate that the current script has hardcoded `WORKDIR` which makes it less practical for a generic use. I believe this issue can be closed.
